frontend/packages/ceph-storage-plugin/src/components/volume-snapshot/volume-snapshot-details.tsx

```diff
@@ -0,0 +1,142 @@
+import './_volume-snapshot.scss';
+
+import * as React from 'react';
+
+import {
+  BreadCrumbs,
+  Firehose,
+  ResourceIcon,
+  ResourceLink,
+  SectionHeading,
+} from '@console/internal/components/utils';
+import { K8sResourceKind, referenceFor } from '@console/internal/module/k8s';
+import { NamespaceModel, PersistentVolumeClaimModel } from '@console/internal/models';
+import { getName, getNamespace } from '@console/shared';
+
+import { VolumeSnapshotModel } from '../../models';
+import { match } from 'react-router-dom';
+
+type VolumeSnapshotDetailsProps = {
+  match: match<{ ns: string; name: string; appName: string }>;
+};
+
+type DetailsComponentProps = {
+  ns: string;
+  pvcName: string;
+  url: string;
+  resources?: {
+    obj: {
+      data: K8sResourceKind;
+    };
+  };
+};
+
+const DetailsComponent: React.FC<DetailsComponentProps> = (props) => {
+  const {
+    ns,
+    pvcName,
+    url,
+    resources: {
+      obj: { data },
+    },
+  } = props;
+  return (
+    <div>
```

Comment:
We need to be using the existing `DetailsPage` component. Look at other details pages for examples